File: projects/order-creation-queries/repository/src/main/java/com/robinfood/repository/configuration/store/IStoreRepository.java
package com.robinfood.repository.configuration.store;

import com.robinfood.core.dtos.configuration.StoreDTO;
import com.robinfood.core.enums.Result;

import java.util.List;

/**
 * Get store configuration by store identifier repository
 */
public interface IStoreRepository {

    /**
     * Get store configuration by store identifier
     *
     * @param storeId The store identifier
     * @param token Token auth service
     * @return Result store
     */
    Result<StoreDTO> getStore(Long storeId, String token);

    /**
     * Get store configuration by stores
     *
     * @param token token auth service
     * @return Result stores
     */
    Result<List<StoreDTO>> getStores(String token);
}

File: projects/posv2/repository/src/main/kotlin/com/robinfood/repository/store/IStoreRepository.kt
package com.robinfood.repository.store

import com.robinfood.core.dtos.store.StoreDeliveryPlatformDTO
import com.robinfood.core.enums.Result

/**
 * Repository for store related data
 */
interface IStoreRepository {

    /**
     * Retrieves the store's delivery platforms
     * [page] page
     * [size] the number of delivery platforms per page
     * [storeId] the store identification to get menu
     * [token] the authentication token to be used
     * @return the active delivery platforms for the store
     */
    suspend fun getStoreDeliveryPlatforms(
            page: Int?,
            size: Int?,
            storeId: Long,
            token: String
    ): Result<List<StoreDeliveryPlatformDTO>>
}